core/runner.py

```diff
@@ -57,14 +57,17 @@ def __init__(self, config_manager):
             JobTypes.JOB_AVG.name: self._average,
             JobTypes.JOB_COMM.name: self._communicate,
         }
+        self.JOBS_NEEDING_STATE = {
+            JobTypes.JOB_COMM.name: self._communicate,
+        }
```

Comment:
In line 79, the callback method always comes from the callbacks dictionary. You only use this dictionary to check if a key is in it. Thus, I think you should just make this an array so that's clearer.

Reply:
Thanks